src/web/releases.rs

```diff
@@ -233,6 +253,35 @@ async fn get_search_results(
     })
 }
 
+// Categorize  errors from registry
+fn handle_registry_error(err: anyhow::Error) -> Result<SearchResult, SearchError> {
```

Comment:
```suggestion
fn handle_registry_error(err: anyhow::Error) -> SearchError {
```

the `Ok` variant will never be returned, right?

Reply:
also, we could put this into from `impl From<anyhow::Error> for SearchError` above?

Reply:
Note: The OK variant will never return. 

Also, I don't feel comfortable with returning `SearchError` directly. While the suggestion makes it simpler, it could lead to issues down the line when the function would fail silently. I'd prefer this to stay explicit.

Reply:
> While the suggestion makes it simpler, it could lead to issues down the line when the function would fail silently.

I don't understand, can you elaborate?

Reply:
I went back and adopted your suggestion, as it makes more sense to return SearchError directly